cmd/syncthing/gui.go

```diff
@@ -369,12 +383,23 @@ func (s *apiService) Serve() {
 		close(s.startedOnce)
 	}
 
-	// Serve in the background
+	// Serve in the background for each listener
 
 	serveError := make(chan error, 1)
-	go func() {
```

Comment:
I don't think the channel needs to be buffered? Can you check thw history why its buffered?

Reply:
Blamed to the following commit:
```
commit 5aacfd16395efc6a375e21113ded6cb3b0d5783e
Author: Jakob Borg <jakob@nym.se>
Date:   Mon Jun 6 22:12:23 2016 +0000

    cmd/syncthing: Make API serve loop more robust (fixes #3136)
    
    This sacrifices the ability to return an error when creating the service
    for being more persistent in keeping it running.
    
    GitHub-Pull-Request: https://github.com/syncthing/syncthing/pull/3270
    LGTM: AudriusButkevicius, canton7
```
The `serveError` channel has been buffered since its birth.

Reply:
@canton7, @calmh remember why?

Reply:
#3136 seems to have a detailed breakdown.

Reply:
That explains the issue, yet I am not sure I understand why it needs to be buffered.

Reply:
I believe that without the 1-buf it is a race between `srv.Serve` returning/sending to `serveError` and control flow in `apiService.Serve` reaching the final select statement where it can read from `serveError`.